framer/Animation.coffee

```
@@ -33,69 +33,48 @@ evaluateRelativeProperty = (target, k, v) ->
 	return target[k] + (sign + 1) * number if sign
 	return +number
 
-# _runningAnimations = []
-
-# Todo: this would normally be BaseClass but the properties keyword
-# is not compatible and causes problems.
 class exports.Animation extends BaseClass
 
-	constructor: (options={}) ->
-
-		options = Defaults.getDefaults "Animation", options
+	constructor: (parameters={},options={}) ->
+		_.defaults(options, parameters.options)
 
-		super options
+		delete parameters.options
+		@options = _.cloneDeep Defaults.getDefaults "Animation", options
+		super parameters
+		@layer = parameters.layer ? null
```

We should throw an error if there is no layer here, and we should make the layer a readonly property.